src/edit.c

```diff
@@ -1687,6 +1687,20 @@ ins_redraw(
 	last_changedtick_buf = curbuf;
 	last_changedtick = CHANGEDTICK(curbuf);
     }
+
+    /* Trigger TextChangedP if b_changedtick differs. */
+    if (ready && has_textchangedP()
+	    && last_changedtick != CHANGEDTICK(curbuf)
+# ifdef FEAT_INS_EXPAND
```

Comment:
If this is not set, the autocmd fires anyway? Probably none of this should fire if ins completion isn’t compiled in?